shared/common-adapters/form-input.native.js

```diff
@@ -0,0 +1,165 @@
+// @flow
+import * as React from 'react'
+import {Box, Text} from '.'
+import type {Props} from './form-input'
+import {NativeTextInput} from './native-wrappers.native'
+import {globalColors, globalMargins, globalStyles} from '../styles'
+import {getStyle} from './text'
+
+type State = {
+  value: string,
+  focused: boolean,
+}
+
+export const backgroundModeToColor = {
+  Normal: globalColors.white,
+  Terminal: globalColors.darkBlue3,
+  Announcements: globalColors.blue,
+  Success: globalColors.green,
+  Information: globalColors.yellow,
+  HighRisk: globalColors.red,
+  Documentation: globalColors.darkBlue,
+}
+
+class FormInput extends React.Component<Props, State> {
+  state: State
+  _input: any
+
+  constructor(props: Props) {
+    super(props)
+
+    this.state = {
+      value: this.props.value || '',
+      focused: false,
+    }
+  }
+
+  getValue = (): string => {
+    return this.state.value || ''
+  }
+
+  setValue = (value: string) => {
+    this.setState({value: value || ''})
+  }
+
+  clearValue = () => {
+    this._onChangeText('')
+  }
+
+  focus = () => {
+    this._input && this._input.focus()
+  }
+
+  blur = () => {
+    this._input && this._input.blur()
+  }
+
+  select = () => {
+    this._input && this._input.select()
+  }
+
+  _onChangeText = (text: string) => {
+    this.setState({value: text || ''})
+
+    this.props.onChangeText && this.props.onChangeText(text || '')
+  }
+
+  _onFocus = () => {
+    this.setState({
+      focused: true,
+    })
+  }
+
+  _onBlur = () => {
+    this.setState({
+      focused: false,
+    })
+  }
+
+  _onKeyPress = (e: SyntheticKeyboardEvent<>) => {
+    if (this.props.onEnterKeyDown && e.key === 'Enter') {
+      this.props.onEnterKeyDown()
+    }
+  }
+
+  render() {
+    let backgroundColor = this.props.backgroundMode
+      ? backgroundModeToColor[this.props.backgroundMode]
+      : globalColors.white
+    return (
+      <Box
+        style={{
```

Comment:
we should chat about not having dynamic styles like this on common components if possible. else it'll always render thrash even if its really not changing. with RN we can use the StyleSheet and make individual styles and you can use the style={[styles.common, this.maxHeight && styles.maxHeight, this.props.containerStyle]} etc